### PR TITLE
feat(sign up): implement users sign up

### PR DESCRIPTION
**Pull request by yvanddniyo**
**What does this PR do?**
  This pull request addresses the implementation of a sign-up method utilizing PostgreSQL as the backend database. The sign-up process allows new users to register for an account within the system.

 **Description of Task to be completed?**
 Here are the endpoint working:
- _GET /api/v1/users_  
-  _POST /api/v1/users/register_  

 **How should this be manually tested?**
1. Clone the repository.
2. Checkout to the branch ft-user-signup-#187431242.
3. Run npm install to install dependencies.
4. Run npm run dev to start the development server.
5. Open your postman or other your testing app and run **http://localhost:3000/api/v1/users** this allow you to retrieve existing user within in the database.
6. Use the endpoint of **http://localhost:3000/api/v1/users/register** to register a user using name, username, email and password.

**What are the relevant pivotal tracker stories?**
#187431242

**Screenshot**

test screenshoot

Screenshot (1)

### CODE REVIEW

**Comment by soleil00** [comment]
your branch ahs conflict did u forget to pull??
ffrom origin dev??

**Review by niyobertin — APPROVED** [review]

File: src/sequelize/config/config.js [hunk 1]
```diff
@@ -5,13 +5,32 @@ module.exports = {
   development: {
     url: process.env.DB_CONNECTION,
     dialect: "postgres",
+    dialectOptions: {
+      ssl: {
+        require: true,
+        rejectUnauthorized: false,
+      },
+    },
```
Comment (teerenzo) [inline src/sequelize/config/config.js]: do you think this can allow us to use local database

File: src/utils/server.ts [hunk 2]
```diff
@@ -1,6 +1,7 @@
 import express from "express";
 import cors from "cors";
 import appROutes from "../routes";
+import userRoutes from "../routes/userRoutes";
```
Comment (teerenzo) [inline src/utils/server.ts]: is this our main router

**Review by teerenzo — CHANGES_REQUESTED** [review]

**Comment by teerenzo** [comment]
work also on your PR naming conversion

File: src/sequelize/config/config.js (hunk 1 above)
Comment (yvanddniyo) [reply src/sequelize/config/config.js]: This allow us to connect to the deployed Database so we are no longer use local database. we're using the deployed one.

File: src/utils/server.ts (hunk 2 above)
Comment (yvanddniyo) [reply src/utils/server.ts]: i fix it

**Comment by teerenzo** [comment]
testing are failing on CI/CD

File: package.json [hunk 3]
```diff
@@ -11,11 +11,13 @@
     "seed": "npx sequelize-cli db:seed:all",
     "lint": "npx eslint .",
     "lint:fix": "npx eslint --fix .",
-    "test": "cross-env NODE_ENV=test jest --detectOpenHandles --coverage"
```
Comment (soleil00) [inline package.json]: bring back cross-env NODE_ENV=test

File: src/sequelize/config/config.js [hunk 4]
```diff
@@ -7,11 +7,18 @@ module.exports = {
     dialect: "postgres",
   },
   test: {
-    url: process.env.TEST_DB,
+    url: process.env.TEST_DB_CONNECTION,
```
Comment (soleil00) [inline src/sequelize/config/config.js]: remove TEST_DB_CONNECTION instead use TEST_DB

File: src/utils/env.ts [hunk 5]
```diff
@@ -4,5 +4,5 @@ dotenv.config();
 export const env = {
   port: process.env.PORT || 3000,
   db_url: process.env.DB_CONNECTION as string,
-  test_db_url: process.env.TEST_DB as string,
+  test_db_url: process.env.TEST_DB_CONNECTION as string,
```
Comment (soleil00) [inline src/utils/env.ts]: same issue remove TEST_DB_CONNECTION use TEST_DB

**Review by soleil00 — CHANGES_REQUESTED** [review]

**Review by Heisjabo — APPROVED** [review]

**Review by soleil00 — APPROVED** [review]

**Review by teerenzo — APPROVED** [review]

**Pull request merged**